tests/integration/test_jenkins.py

```diff
@@ -55,3 +59,41 @@ async def test_jenkins_automatic_update_out_of_range(
     assert unit_web_client.client.has_plugin(
         extra_plugin
     ), "additionally installed plugin cleanedup."
+
+
+async def test_storage_mount(
+    application: Application,
+    jenkins_client: jenkinsapi.jenkins.Jenkins,
+):
+    """
+    arrange: a bare Jenkins charm.
+    act: Add a job, scale the charm to 0 unit and scale back to 1.
+    assert: The job configuration persists and is the same as the one used.
+    """
+    test_job_name = token_hex(8)
+    job_configuration = gen_test_job_xml("built-in")
+    jenkins_client.create_job(test_job_name, job_configuration)
+
+    await application.scale(scale=0)
+    await application.model.wait_for_idle(
+        apps=[application.name],
+        timeout=20 * 60,
+        idle_period=30,
+        wait_for_exact_units=0,
+    )
+    await application.scale(scale=1)
+    await application.model.wait_for_idle(
+        apps=[application.name],
+        timeout=20 * 60,
+        idle_period=30,
+        wait_for_exact_units=1,
+    )
+
+    jenkins_unit: Unit = application.units[0]
+    assert jenkins_unit
+    command = f"cat /var/lib/jenkins/jobs/{test_job_name}/config.xml"
```

Comment:
`jenkins_unit.get_container("jenkins").pull_path` should simplify this

Reply:
I think that works for unit testing harness but not for integration test juju.Unit unfortunately

Reply:
I just tested locally and it doesn't seems like `.get_container` is supported on `juju.Unit` with juju version pinned in tox (3.0.4)

`file = jenkins_unit.get_container("jenkins").pull_path(
        f"/var/lib/jenkins/jobs/{test_job_name}/config.xml"
    )`

Gives me:
```
========================================================================================= short test summary info ==========================================================================================
FAILED tests/integration/test_jenkins.py::test_storage_mount - KeyError: 'get_container'

```

Reply:
I checked with 3.2.3.0 on python-libjuju but it still doesn't seem like the method is implemented. I'll try looking at other repos to see if this is used anywhere